tax-processing-functions/quality_and_split/main.py

```diff
@@ -0,0 +1,403 @@
+    # input_bucket_name = GCS bucket to copy final processed doc to
+    if raw_content:
+        for entity in raw_content.entities:
+            if "1040sc" in str(entity.type):
+                # move to 1040c input bucket
+                move_blob(input_bucket_name, bucket_1040c_input, blob_name=file_name)
+
+                # extract and format entities from document
+                extracted_1040_doc = extract_entities(
+                    file, content_type, processor_1040
+                )  # parse 1040c doc
+                extracted_1040_doc_processed = format_entities(
+                    extracted_1040_doc
+                )  # formatted
+                json_string = json.dumps(extracted_1040_doc_processed, sort_keys=False)
+
+                # Get timestamp and upload to GCS
+                stamp = datetime.now()
+                str_date_time = stamp.strftime(ts_format)
+                upload_blob(
+                    bucket_1040c_processed,
+                    f"1040sc_{str_date_time}_{file_name}",
+                    json_string,
+                )
+
+                # Write formatted data to BQ Table
+                write_to_bq(
+                    project_name,
+                    dataset_name,
+                    table_name,
+                    extracted_1040_doc_processed,
+                    file_name,
+                    "1040sc",
+                )
+                print("1040sc output has been written to BQ.")
+
+                # Publish to pubsub
+                topic_path = publisher.topic_path(project_id, topic_name)
+                publish_to_pubsub(topic_path, json_string)
+
+            elif "1120s" in str(entity.type):
+                # move to 1040c input bucket
+                move_blob(input_bucket_name, bucket_1120s_input, blob_name=file_name)
+
+                # extract and format entities from document
+                extracted_1120_doc = extract_entities(
+                    file, content_type, processor_1120
+                )  # parse 1120s doc
+                extracted_1120_doc_processed = format_entities(
+                    extracted_1120_doc
+                )  # formatted
+                json_string = json.dumps(extracted_1120_doc_processed, sort_keys=False)
+
+                # Get timestamp and upload to GCS
+                stamp = datetime.now()
+                str_date_time = stamp.strftime(ts_format)
+                upload_blob(
+                    bucket_1120s_processed,
+                    f"1120sk_{str_date_time}_{file_name}",
+                    json_string,
+                )
+
+                # Write formatted data to BQ Table
+                write_to_bq(
+                    project_name,
+                    dataset_name,
+                    table_name,
+                    extracted_1120_doc_processed,
+                    file_name,
+                    "1120ssk1",
+                )
+                print("1120s output has been written to BQ.")
+
```

Comment:
A lot of this code looks to be duplicated. Might be worth pulling out into a separate function or seeing where you can move code outside the if statements

Reply:
Don't have time today to try this, but can look at it later if it is needed